pkg/haprovider/haprovider.go

```diff
@@ -364,13 +376,24 @@ func (r *HAProvider) CreateOrUpdateHAEndpoints(ctx context.Context, machine *clu
 		return err
 	}
 
+	// Get control plane endpoint ip family
+	adcForCluster, err := r.getADCForCluster(ctx, cluster)
+	if err != nil {
+		r.log.Error(err, "Failed to get cluster AKODeploymentConfig")
+		return err
+	}
+	ipFamily := "V4"
+	if adcForCluster != nil {
+		ipFamily = utils.GetIPFamilyFromCidr(adcForCluster.Spec.DataNetwork.CIDR)
```

Comment:
This is incorrect, we should get ako's ipfamily configure, not VIP network IP type, so it should get from adc.spec.extraconfig.ipfamily.

Reply:
This part is to decide the ip type of control plane endpoint ip, it should stay same as the ip type of VIP/data network cidr

Reply:
You are using this ipfamily to select the backend machine, so it should stay the same as the IP type of the AKO's ipfamily?